nodestream/pipeline/extractors/queues/sqs.py

```diff
@@ -0,0 +1,86 @@
+import asyncio
+from logging import getLogger
+from typing import Any, AsyncGenerator, Iterable, Optional
+
+from nodestream.pipeline.extractors.queues.extractor import QueueConnector
+
+from ..credential_utils import AwsClientFactory
+
+
+class SQSQueueConnector(QueueConnector, alias="sqs"):
+    @classmethod
+    def from_file_data(
+        cls,
+        queue_url: str,
+        message_system_attribute_names: Optional[list[str]] = ["All"],
+        message_attribute_names: Optional[list[str]] = ["All"],
+        max_batch_size: int = 10,
+        delete_after_read: bool = True,
+        max_batches: int = 10,
+        **aws_client_args,
+    ):
+        return cls(
+            queue_url=queue_url,
+            message_system_attribute_names=message_system_attribute_names,
+            message_attribute_names=message_attribute_names,
+            max_batch_size=max_batch_size,
+            delete_after_read=delete_after_read,
+            max_batches=max_batches,
+            sqs_client=AwsClientFactory(**aws_client_args).make_client("sqs"),
+        )
+
+    def __init__(
+        self,
+        sqs_client,
+        queue_url: str,
+        message_system_attribute_names: Optional[list[str]] = ["All"],
+        message_attribute_names: Optional[list[str]] = ["All"],
+        max_batch_size: int = 10,
+        delete_after_read: bool = True,
+        max_batches: int = 10,
+    ) -> None:
+        self.queue_url = queue_url
+        self.message_system_attribute_names = message_system_attribute_names
+        self.message_attribute_names = message_attribute_names
+        self.max_batch_size = max_batch_size
+        self.delete_after_read = delete_after_read
+        self.max_batches = max_batches
+        self.sqs_client = sqs_client
+        self.logger = getLogger(__name__)
+
+    async def poll(self) -> Iterable[Any]:
+        results = []
+        for _ in range(self.max_batches):
+            try:
+                messages = await self.get_next_messsage_batch()
+                if messages is None:
+                    self.logger.debug("Polling returned no messages")
+                    continue
+                for result in messages:
+                    results.append(result)
+            except Exception as e:
+                self.logger.exception(f"error while polling SQS messages: {e}")
+                break
+        return results
+
+    async def get_next_messsage_batch(self):
+        loop = asyncio.get_running_loop()
+        msgs = await loop.run_in_executor(None, self.get_message_batch)
+        messages = msgs.get("Messages")
+        if messages is None:
+            return None
+        return self.process_messages(messages)
```

Comment:
I am None and boolean averse. I think we can achieve the same behavior and simply out this if statement by having `msgs.get("Messages")` default to an empty list with `msgs.get("Messages", [])` and then when we check if `if messages is None`, simply check if `len(messages) ==. 0`. This removes the conditional and changes the type from list | none to just list. We can also then drop the continue simplifying the loop.